src/spiral_stc.cpp

```diff
@@ -8,294 +8,356 @@
-      dx = -dy;
-      dy = dx_prev;
+      // Get node from parent
+      node_ = parent.lock();
+      name_ = name;
+
+      // Currently this plugin does not use the costmap, instead request a map from a server
+      // This will change in the future
+      costmap_ = costmap_ros->getCostmap();
+      global_frame_ = costmap_ros->getGlobalFrameID();
+
+      RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"),
+                  "Configuring plugin %s of type NavfnPlanner", name_.c_str());
+
+      // Create a publisher to visualize the plan
+      plan_pub_ = node_->create_publisher<nav_msgs::msg::Path>("plan", 1);
+      // Try to request the cpp-grid from the cpp_grid map_server
+      cpp_grid_client_ = node_->create_client<nav_msgs::srv::GetMap>("/map_server/map");
+
+      // Define  robot radius (radius) parameter
+      float robot_radius_default = 0.5f;
+      declare_parameter_if_not_declared(node_, name_ + ".robot_radius", rclcpp::ParameterValue(robot_radius_default));
+      node_->get_parameter(name_ + ".robot_radius", robot_radius_);
+      // Define  tool radius (radius) parameter
+      float tool_radius_default = 0.5f;
+      declare_parameter_if_not_declared(node_, name_ + ".tool_radius", rclcpp::ParameterValue(tool_radius_default));
+      node_->get_parameter(name_ + ".tool_radius", tool_radius_);
+      initialized_ = true;
     }
-    else
+  }
+
+  void SpiralSTC::activate()
+  {
+    RCLCPP_INFO(
+      rclcpp::get_logger("FullCoveragePathPlanner"), "Activating plugin %s of type FullCoveragePathPlanner",
+      name_.c_str());
+  }
+
+  void SpiralSTC::deactivate()
+  {
+    RCLCPP_INFO(
+      rclcpp::get_logger("FullCoveragePathPlanner"), "Deactivating plugin %s of type FullCoveragePathPlanner",
+      name_.c_str());
+  }
+
+  void SpiralSTC::cleanup()
+  {
+    RCLCPP_INFO(
+      rclcpp::get_logger("FullCoveragePathPlanner"), "Cleaning up plugin %s of type FullCoveragePathPlanner",
+      name_.c_str());
+    // TODO(clopez) Add proper cleanup
+  }
+
+  nav_msgs::msg::Path SpiralSTC::createPlan(
+    const geometry_msgs::msg::PoseStamped & start,
+    const geometry_msgs::msg::PoseStamped & goal)
+  {
+    nav_msgs::msg::Path global_path;
+    std::vector<geometry_msgs::msg::PoseStamped> plan;
+
+    SpiralSTC::makePlan(start, goal, plan);
+
+    global_path.poses.clear();
+    global_path.header.stamp = node_->now();
+    global_path.header.frame_id = global_frame_;
+    for (auto pose : plan)
     {
-      // Initialize spiral direction towards y-axis
-      dx = 0;
-      dy = 1;
+      global_path.poses.push_back(pose);
     }
-    done = true;
-
-    for (int i = 0; i < 4; ++i)
+    return global_path;
+  }
+  std::list<gridNode_t> SpiralSTC::spiral(std::vector<std::vector<bool>> const &grid, std::list<gridNode_t> &init,
+                                          std::vector<std::vector<bool>> &visited)
+  {
+    int dx, dy, dx_prev, x2, y2, nRows = grid.size(), nCols = grid[0].size();
```

Comment:
`dx` is not used outside the while loop. So declare it there.
Also if you declare it as `int dx=0;` you don't have the first-loop declaration and you prevent the variable from being accidentally used un-initialised.

Same holds for `dy` and probably others.